tools/localization/localeTypes.py

```diff
@@ -1,10 +1,44 @@
 #!/bin/python3
 import re
-from typing import List, Tuple
 
 OUTPUT_FILE = "./ts/localization/locales.ts"
 
 
+# Define the mapping of special fields to their types and corresponding "WithX"
+with_map = {
+    ("name", "string"): "WithName",
```

Comment:
Lol this fill be fun to maintain. But good work coming up with a solution.

No idea if this would be possible but would be nice to have these linked to somesort of dropdown list in crowdin when creating new strings and selecting arugments. It would give us some degree of type safety 👀